File: Jasmine/JasmineCompiler/Src/DFIR_Compiler/DFIR_Compiler/ICompoundArithmeticNode.cs
using System;
using System.Collections.Generic;
using System.Xml;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace DFIR_Compiler
{
    public class ICompoundArithmeticNode:INode
    {
        private ICompoundArithmeticNodeMode mode;
        private List<bool> invertedInputs;
        private bool invertedOutput;
        public ICompoundArithmeticNode(int nodeId, int parentId, List<ITerminal> inputTerminals, List<ITerminal> outputTerminals, ICompoundArithmeticNodeMode Mode, XmlNodeList InvertedInputs, XmlNode InvertedOutput) : base(nodeId, parentId, inputTerminals, outputTerminals)
        {
            mode = Mode;
            invertedInputs = new List<bool>();
            for (int i = 0; i < InvertedInputs.Count; i++)
            {
                //Set the inverted conditions for input terminals
                if (InvertedInputs[i].InnerText.Equals("False"))
                {
                    invertedInputs.Add(false);
                }
                else
                {
                    invertedInputs.Add(true);
                }
            }
            //Set the inverted condition for output terminal
            if (InvertedOutput.InnerText.Equals("False"))
            {
                invertedOutput = false;
            }
            else
            {
                invertedOutput = true;
            }
            INodeType = "ICompoundArithmeticNode";
        }


        public virtual ICompoundArithmeticNodeMode GetMode()
        {
            return mode;
        }
        public virtual List<bool> GetInvertedInputs()
        {
            return invertedInputs;
        }
        public virtual bool GetInvertedOutput()
        {
            return invertedOutput;
        }


        public override void Print()
        {
            Console.WriteLine("------------------------------------");
            Console.WriteLine("ICompoundArithmeticNode:");
            Console.WriteLine("NodeId: " + nodeId + ' ' + "ParentId: " + parentId + "\nMode: " + mode);
            for (int i = 0; i < invertedInputs.Count; i++)
            {
                Console.WriteLine("InvertedInput: " + invertedInputs[i]);
            }
            Console.WriteLine("InvertedOutput: " + invertedOutput);
            for (int i = 0; i < inputTerminals.Count; i++)
            {
                inputTerminals[i].Print();
            }
            for (int i = 0; i < outputTerminals.Count; i++)
            {
                outputTerminals[i].Print();
            }
        }

    }
}

File: Wayne/WayneCompiler/src/Dfir_representation/ICompoundArithmeticNode.java
package Dfir_representation;

import java.util.Vector;

import org.w3c.dom.Node;
import org.w3c.dom.NodeList;


public class ICompoundArithmeticNode extends JNode{
	   
	   private ICompoundArithmeticNodeMode mode;
	   private Vector<Boolean> invertedInputs;
	   private Boolean invertedOutput;
	   public ICompoundArithmeticNode(int nodeId, int parentId, Vector<ITerminal> inputTerminals,Vector<ITerminal> outputTerminals, ICompoundArithmeticNodeMode mode, NodeList InvertedInputs, Node InvertedOutput){
		   super(nodeId,parentId,inputTerminals,outputTerminals);
		   this.mode = mode;
		   this.invertedInputs = new Vector<Boolean>();
		   for(int i=0; i<InvertedInputs.getLength(); i++){
			   if(InvertedInputs.item(i).getTextContent().equals("False")){
				   this.invertedInputs.addElement(false);
			   }
			   else{
				   this.invertedInputs.addElement(true);
			   }
		   }
		   if(InvertedOutput.getTextContent().equals("False")){
			   this.invertedOutput = false;
		   }
		   else{
			   this.invertedOutput = true;
		   }
		   JNodeType = "ICompoundArithmeticNode";
	   }
	   public ICompoundArithmeticNodeMode getMode(){
		   return this.mode;
	   }
	   public Vector<Boolean> getInvertedInputs(){
		   return this.invertedInputs;
	   }
	   
	   public Boolean getInvertedOutput(){
		   return this.invertedOutput;
	   }
	   public void print(){
		   System.out.println("ICompoundArithmeticNode begin");
		   System.out.println("NodeId: " + this.nodeId + ' ' + "ParentId: " + this.parentId + "\nMode: " + this.mode);
		   for(int i=0; i<invertedInputs.size(); i++){
			   System.out.println("InvertedInput: " + invertedInputs.get(i));
	       }
		   System.out.println("InvertedOutput: " + invertedOutput);
		   for(int i=0; i<inputTerminals.size(); i++){
				   inputTerminals.get(i).print();
		   }
		  for(int i=0; i<outputTerminals.size(); i++){
				   outputTerminals.get(i).print();
		   }   
		  System.out.println("ICompoundArithmeticNode end\n");
	   }
}